src/consensus/vote.cpp

```diff
@@ -86,11 +86,26 @@ VoteManager::VoteManager(addr_t node_addr, std::shared_ptr<DbStorage> db, std::s
     : db_(db), pbft_chain_(pbft_chain), final_chain_(final_chain) {
   LOG_OBJECTS_CREATE("VOTE_MGR");
   // Retrieve votes from DB
+  daemon_ = std::make_unique<std::thread>([this]() { retreieveVotes_(); });
+
+  current_period_final_chain_block_hash_ = final_chain_->block_header()->hash;
+}
+
+VoteManager::~VoteManager() { daemon_->join(); }
```

Comment:
I think we can stop the daemon by end of retrieveVotes_()

Reply:
That would be in the same thread..??